src/indra_cogex/client/queries.py

```diff
@@ -1740,6 +1794,1447 @@ def _filter_out_medscan_evidence(
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    trial : Tuple[str, str]
+        The clinical trial to query (e.g., ("clinicaltrials", "NCT00201240"))
+
+    Returns
+    -------
+    :
+        Research project nodes associated with this clinical trial
+    """
+    return client.get_sources(
+        trial,
+        relation="has_clinical_trial",
+        source_type="ResearchProject",
+        target_type="ClinicalTrial"
+    )
+
+
+@autoclient()
+def get_patents_for_project(
+    project: Tuple[str, str], *, client: Neo4jClient
+) -> Iterable[Node]:
+    """Return patents associated with an NIH research project.
+
+    Parameters
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    project : Tuple[str, str]
+        The project to query (e.g., ("nihreporter.project", "2106676"))
+
+    Returns
+    -------
+    :
+        Patent nodes associated with this project
+    """
+    return client.get_targets(
+        project,
+        relation="has_patent",
+        source_type="ResearchProject",
+        target_type="Patent"
+    )
+
+
+@autoclient()
+def get_projects_for_patent(
+    patent: Tuple[str, str], *, client: Neo4jClient
+) -> Iterable[Node]:
+    """Return NIH research projects associated with a patent.
+
+    Parameters
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    patent : Tuple[str, str]
+        The patent to query (e.g., ("google.patent", "US5939275"))
+
+    Returns
+    -------
+    :
+        Research project nodes associated with this patent
+    """
+    return client.get_sources(
+        patent,
+        relation="has_patent",
+        source_type="ResearchProject",
+        target_type="Patent"
+    )
+
+
+#interpro
+@autoclient()
+def get_domains_for_gene(
+    gene: Tuple[str, str], *, client: Neo4jClient
+) -> Iterable[Node]:
+    """Return protein domains associated with the given gene.
+
+    Parameters
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    gene : Tuple[str, str]
+        The gene to query (e.g., ("hgnc", "475"))
+
+    Returns
+    -------
+    :
+        Domain nodes (InterPro) associated with this gene
+    """
+    return client.get_targets(
+        gene,
+        relation="has_domain",
+        source_type="BioEntity",
+        target_type="BioEntity"
+    )
+
+
+@autoclient()
+def get_genes_for_domain(
+    domain: Tuple[str, str], *, client: Neo4jClient
+) -> Iterable[Node]:
+    """Return genes associated with the given protein domain.
+
+    Parameters
+    ----------
+    client : Neo4jClient
+        The Neo4j client
+    domain : Tuple[str, str]
+        The domain to query (e.g., ("interpro", "IPR006047"))
+
+    Returns
+    -------
+    :
+        Gene nodes (HGNC) associated with this domain
+    """
+    return client.get_sources(
+        domain,
+        relation="has_domain",
+        source_type="BioEntity",
+        target_type="BioEntity"
+    )
+
+
+@autoclient()
+def has_domain(
```

Comment:
I think we can get a better name than `has_domain`, maybe `gene_has_domain`?

Reply:
Fixed in 4640b451